sdk/core/core-paging/src/getPagedAsyncIterator.ts

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  PagedAsyncIterableIterator,
+  PageSettings,
+  PagedResult,
+  PagedAsyncIteratorOptions
+} from "./models";
+
+/**
+ * returns an async iterator that will retrieve items from the server. It also has a `byPage`
+ * method that can return pages of items at once.
+ *
+ * @param pagedResult - an object that describes how to communicate with the service and how to build a page of items.
+ * @param path - the path to the resource to retrieve
+ * @param getRequestOptions - the options to pass to the service
+ * @returns a paged async iterator that will retrieve items from the server.
+ */
+export function getPagedAsyncIterator<
+  TFetchPageOptions,
+  TResponse,
+  TElement,
+  TPage = TElement[],
+  TPageSettings = PageSettings
+>(
+  pagedResult: PagedResult<TFetchPageOptions, TResponse, TPage>,
+  path: string,
+  getRequestOptions: TFetchPageOptions,
+  options?: PagedAsyncIteratorOptions<TResponse, TPage, TPageSettings>
+): PagedAsyncIterableIterator<TElement, TPage, TPageSettings> {
+  const iter = getItemAsyncIterator<TFetchPageOptions, TResponse, TElement, TPage, TPageSettings>(
+    pagedResult,
+    path,
+    getRequestOptions,
+    options
+  );
+  return {
+    next() {
+      return iter.next();
+    },
+    [Symbol.asyncIterator]() {
+      return this;
+    },
+    byPage:
+      options?.byPage ??
+      ((settings?: PageSettings) => {
+        const pageOptions = { ...getRequestOptions };
+        (pageOptions as Record<string, unknown>)[options?.maxPageSizeParam ?? "top"] =
+          settings?.maxPageSize;
+        return getPageAsyncIterator(
+          pagedResult,
+          settings?.continuationToken ?? path,
```

Comment:
I am not sure what is the intended usage for the `continuationToken` parameter so I am using it as a custom url for nextLink. @xirzec, @sarangan12, @joheredi thoughts?

Reply:
In general it depends on the service. Sometimes it is an opaque value appended as a query string (with a service-specific query string parameter). It can also be passed in as a header. Sometimes it can be multiple items, e.g. I believe azure tables has a separate key for the next partition and the next row. I suspect that will need to allow clients to decide how a continuation token is used using code rather than configuration?

Reply:
@bterlson actually that was helpful! based on what you said, I think we should not have a default interpretation/implementation for it. It is pretty much service and API specific. I am ignoring it now in the default `byPage`.